questionnaire/routes.test.js

```diff
@@ -628,30 +652,121 @@ describe('Hitting /apply/:section', () => {
                     './form-helper.js': jest.fn(() => ({
                         getSectionContext: () => 'submission',
                         addPrefix: section => `p-${section}`,
-                        processRequest: requestBody => requestBody
+                        processRequest: requestBody => requestBody,
+                        removeSectionIdPrefix: () => 'applicant-fatal-claim'
                     }))
                 });
             });
             it('Should submit', async () => {
                 const currentAgent = request.agent(app);
                 const initialResponse = await currentAgent.get('/apply/start-or-resume');
                 const initialCsrfToken = getCsrfTokenFromResponse(initialResponse.res.text);
+                const initialExternalId = 'urn:uuid:ce66be9d-5880-4559-9a93-df15928be396';
                 const response = await currentAgent.post('/apply/applicant-fatal-claim').send({
-                    _csrf: initialCsrfToken
+                    _csrf: initialCsrfToken,
+                    _externalId: initialExternalId
                 });
                 expect(response.res.text).toContain('Application submitted');
             });
             it('Should redirect to you have submitted your application page on refresh after confirmation page', async () => {
                 const currentAgent = request.agent(app);
                 const initialResponse = await currentAgent.get('/apply/start-or-resume');
                 const initialCsrfToken = getCsrfTokenFromResponse(initialResponse.res.text);
+                const initialExternalId = 'urn:uuid:ce66be9d-5880-4559-9a93-df15928be396';
+
                 const response = await currentAgent.post('/apply/applicant-fatal-claim').send({
-                    _csrf: initialCsrfToken
+                    _csrf: initialCsrfToken,
+                    _externalId: initialExternalId
                 });
                 expect(response.res.text).toContain('Application submitted');
             });
         });
-
+        describe('Prevent cross-template edits', () => {
+            describe('Authenticated', () => {
+                beforeEach(() => {
+                    setUpCommonMocks({
+                        './form-helper.js': jest.fn(() => ({
+                            getSectionContext: () => 'post',
+                            addPrefix: section => `p-${section}`,
+                            processRequest: requestBody => requestBody,
+                            removeSectionIdPrefix: () => 'applicant-fatal-claim'
+                        })),
+                        '../account/account-service': () => {
+                            return jest.fn(() => ({
+                                getOwnerId: () => 'urn:uuid:625cc31a-dc57-4064-860e-e68d049035ad',
+                                isAuthenticated: () => true
```

Comment:
I do not know why this wasn't working when I first wrote it but I tested it this morning and now it is